packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -61,14 +72,19 @@ class CupertinoCheckbox extends StatefulWidget {
     required this.value,
     this.tristate = false,
     required this.onChanged,
+    this.mouseCursor,
     this.activeColor,
     this.inactiveColor,
```

Comment:
Yes, `Checkbox` does that with `MaterialStateProperty`. My guess is `Checkbox` and `CupertinoCheckbox`'s API's were designed before `MaterialStateProperty` could configure `fillColor` in both active and inactive states, and so `activeColor` and `inactiveColor` aren't needed anymore in both `Checkbox` and `CupertinoCheckbox`. Both parameters could (and probably should) be removed, but wouldn't that break users?

Reply:
It could be breaking, but we typically will deprecate and maintain backwards compatibility for a while so folks can take the time to migrate to the new API. Since CupertinoSwitch uses the same active/inactive parameters, I think this is OK. @MitchellGoodwin had some really good thoughts on how to evaluate when we do or don't use WSP here: https://github.com/flutter/flutter/pull/149465#issuecomment-2150489211

Reply:
Currently inactiveColor only changes the border color when the checkbox is unselected. Which the changes to shape in this PR make that a little redundant. I think `inactiveColor` should be deprecated, because now it just doesn't do much. There's an argument to deprecate `activeColor` as well, but I think it's fine to keep it around for the simple cases.

Reply:
Filed tracking issue https://github.com/flutter/flutter/issues/151252